### PR TITLE
Add Cons, Last, Init and Elem

### PR DESCRIPTION
**Pull request by blmage**

### CODE REVIEW

**Comment by blmage** [comment]
This should be better now!

File: src/Fcf/Data.hs [hunk 1]
```diff
@@ -129,6 +136,17 @@ data Head :: [a] -> Exp (Maybe a)
 type instance Eval (Head '[]) = 'Nothing
 type instance Eval (Head (a ': _as)) = 'Just a
 
+data Last :: [a] -> Exp (Maybe a)
+type instance Eval (Last '[]) = 'Nothing
+type instance Eval (Last (a ': as)) =
+  Eval (If (Eval (Null as)) (Pure ('Just a)) (Last as))
```
Comment (Lysxia) [inline src/Fcf/Data.hs]: You could also do a similar case split as for `Init` here.

Comment (blmage) [reply src/Fcf/Data.hs]: Thanks for your quick feedbacks!

**Comment by Lysxia** [comment]
Thanks!

**Pull request merged**